torch/utils/_python_dispatch.py

```diff
@@ -85,28 +85,28 @@ def _get_current_dispatch_mode_stack():
     return [_get_dispatch_stack_at(i) for i in range(stack_len)]
 
 def _push_mode(mode, k: Optional[DispatchKey] = None):
-    if k is not None:
-        from torch._ops import push_mode_for_key, get_cached_ops
-        # See Note [Not Caching Per-Dispatch-Key Mode Handlers]
-        # Clear the cache of every op that has been used so far, for this particular key.
-        ks = torch._C._functionality_to_backend_keys(k)
-        for op in get_cached_ops():
-            for key in ks:
-                op._uncache_dispatch(key)
-        push_mode_for_key(k, mode)
-    else:
+    assert k is None or k == torch._C.DispatchKey.PreDispatch
```

Comment:
Since we're killing "per-dispatch-key-modes" in this PR, we should also probably just change the API to accept a `bool` instead of an arbitrary dispatch key, and also change `ProxyTorchDispatchMode` so it stashes a bool instead of a dispatch key. If you don't want to do that as part of this PR though np